tests/test_set_config.py

```diff
@@ -405,3 +405,123 @@ def test_values_needed_f90nml():
     salad.appetizer
 """
     assert result == outcome
+
+def test_cfg_to_yaml_conversion():
+    ''' Test that .a cfg file can be used to create a yaml object.'''
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2.cfg"))
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        args = ['-i', input_file, '--dry_run', '--input_file_type', '.yaml']
+
+        outstring = io.StringIO()
+        with redirect_stdout(outstring):
+            set_config.create_config_obj(args)
+        result = outstring.getvalue()
+
+        expected = config.YAMLConfig(input_file)
+        expected_file = f'{tmp_dir}/test.yaml'
+        expected.dump_file(expected_file)
+
+        assert result.rstrip('\n') == str(expected)
+
+def test_output_file_conversion():
```

Comment:
def test_output_file_conversion(): is where we're seeing the MissingSectionHeader error